codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/serialize/JsonSerializerGenerator.kt

```diff
@@ -407,11 +405,7 @@ class JsonSerializerGenerator(
                 val timestampFormat =
                     httpBindingResolver.timestampFormat(context.shape, HttpLocation.DOCUMENT, EPOCH_SECONDS)
                 val timestampFormatType = RuntimeType.timestampFormat(runtimeConfig, timestampFormat)
-                rustTemplate(
-                    "$writer.date_time(${value.asRef()}#{ConvertInto:W}, #{FormatType})?;",
-                    "FormatType" to timestampFormatType,
-                    "ConvertInto" to typeConversionGenerator.convertViaInto(target),
-                )
+                rust("$writer.date_time(${value.asRef()}, #T)?;", timestampFormatType)
```

Comment:
I've reverted the changes made in https://github.com/awslabs/smithy-rs/commit/5455c4e9cce288f57367176d8e1c64892842167f#diff-41136e956b536306dde14246dbe368a0450d6849773cc95f14f72685f0e3b022 to fix `Timestamp` conversions errors from Python counterpart. We're now using a `Deref` hack to accomplish that because current solution wasn't enough for some cases. `Deref` hack is explained in https://github.com/awslabs/smithy-rs/blob/843d1549d018d21c2e1c8b5fe9f91ff27a09a81d/rust-runtime/aws-smithy-http-server-python/src/types.rs#L8-L16